### DISCUSSION TITLE
Suggestion to quickly checkout when in Local tree directory

### DISCUSSION THREAD

**Discussion by leonandroid** [gitextensions/gitextensions#8792] [Ideas]
Hello.

Due to the role I have where I work, I found usually the need to swap between branches more often than usual, and the amount of branches that my team has is quite big, more than 400, but the amount that I have locally checked out is never greater than 10. So when I need to fetch latest changes and checkout a branch, I need to search for the branch in the origin nodes, which takes me a lot of time.

I would LOVE to have the option in the Local Branches, when you right click the branch in the branch explorer (left side directory tree), when the branch is outdated (showing amount of commits behind) a new option saying "Fetch and Checkout". It should do the same thing that does when you click that option in the Origin section.

![ideaForGitEx](https://user-images.githubusercontent.com/1564651/105619398-1eba3580-5dd1-11eb-8f23-7707de83ca19.png)

As the local branch has the reference to the origin branch, I suppose this is not something impossible to achieve.

Hope my suggestion helps.
Regards

**Comment by vbjay** [comment]
It would have to be a local branch that has a tracking branch reference.
But should be doable. If only local, no menu or disable menu.

**Comment by gerhardol** [comment]
or a fetch dialog similar to push that lists local and remotes (also those where name matches but not tracking)
See master for improvements that shows ahead/behind
